x/auction/simulation/operations/msg.go

```diff
@@ -0,0 +1,180 @@
+package operations
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	authexported "github.com/cosmos/cosmos-sdk/x/auth/exported"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/kava-labs/kava/x/auction"
+	"github.com/kava-labs/kava/x/auction/keeper"
+	"github.com/kava-labs/kava/x/auction/types"
+)
+
+var (
+	noOpMsg = simulation.NoOpMsg(auction.ModuleName)
+)
+
+// Return a function that runs a random state change on the module keeper.
+// There's two error paths
+// - return a OpMessage, but nil error - this will log a message but keep running the simulation
+// - return an error - this will stop the simulation
+func SimulateMsgPlaceBid(authKeeper auth.AccountKeeper, keeper keeper.Keeper) simulation.Operation {
+	handler := auction.NewHandler(keeper)
+
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
+		simulation.OperationMsg, []simulation.FutureOperation, error) {
+
+		// get open auctions
+		openAuctions := types.Auctions{}
+		keeper.IterateAuctions(ctx, func(a types.Auction) bool {
+			openAuctions = append(openAuctions, a)
+			return false
+		})
+
+		// shuffle auctions slice so that bids are evenly distributed across auctions
+		rand.Shuffle(len(openAuctions), func(i, j int) {
+			openAuctions[i], openAuctions[j] = openAuctions[j], openAuctions[i]
+		})
+		// TODO do the same for accounts?
+		var accounts []authexported.Account
+		for _, acc := range accs {
+			accounts = append(accounts, authKeeper.GetAccount(ctx, acc.Address))
+		}
+
+		// search through auctions and an accounts to find a pair where a bid can be placed (ie account has enough coins to place bid on auction)
+		blockTime := ctx.BlockHeader().Time
+		bidder, auction, found := findValidAccountAuctionPair(accounts, openAuctions, func(acc authexported.Account, auc types.Auction) bool {
+			_, err := generateBidAmount(r, auc, acc, blockTime)
+			if err == ErrorNotEnoughCoins {
+				return false // keep searching
+			} else if err != nil {
+				panic(err) // raise errors
+			}
+			return true // found valid pair
+		})
+		if !found {
+			return noOpMsg, nil, nil // TODO op message?
```

Comment:
I think we should log when this occurs - I would be curious to know the frequency and to inspect the state when it does. Maybe:
```suggestion
			return NewOperationMsgBasic(auction.ModuleName, "no-operation (could not find bidder)", "", false, nil)
```

Reply:
As suggested, I renamed the OpMsg "name" to clarify the reason for there being no operation.
I also added the error log from submitting msgs to the "comment" field of the op msgs to aid in debugging.